src/Microsoft.Health.Fhir.TemplateManagement/Client/OrasClient.cs

```diff
@@ -0,0 +1,107 @@
+using Microsoft.Health.Fhir.TemplateManagement.Models;
+
+namespace Microsoft.Health.Fhir.TemplateManagement.Client
+{
+    // Todo need to be async
+
+    public class OrasClient : IOrasClient
+    {
+        private readonly string _imageReference;
+
+        public OrasClient(string imageReference)
+        {
+            EnsureArg.IsNotNull(imageReference, nameof(imageReference));
+
+            _imageReference = imageReference;
+        }
+
+        public async Task<bool> PullImageAsync(string outputFolder)
+        {
+            string command = $"pull  {_imageReference} -o {outputFolder}";
+            await OrasExecutionAsync(command, ".");
+            return true;
+        }
+
+        public async Task<bool> PushImageAsync(string inputFolder)
+        {
+            string argument = string.Empty;
+            string command = $"push {_imageReference}";
+
+            if (!Directory.Exists(inputFolder))
+            {
+                Console.WriteLine($"No file for push.");
+                return false;
+            }
+
+            var filePathToPush = Directory.EnumerateFiles(inputFolder, "*.tar.gz", SearchOption.AllDirectories);
+            foreach (var filePath in filePathToPush)
+            {
+                argument += $" {Path.GetRelativePath(inputFolder, filePath)}";
+            }
+
+            if (string.IsNullOrEmpty(argument))
+            {
+                Console.WriteLine($"No file for push.");
+                return false;
+            }
+
+            // In order to remove image's directory prefix. (e.g. "layers/layer1.tar.gz" --> "layer1.tar.gz"
+            // Change oras working folder into imageLayerFolder
+            await OrasExecutionAsync(string.Concat(command, argument), inputFolder);
+            return true;
+        }
+
+        private async Task OrasExecutionAsync(string command, string orasWorkingDirectory)
+        {
+            TaskCompletionSource<bool> eventHandled = new TaskCompletionSource<bool>();
+            Process process = new Process
+            {
+                StartInfo = new ProcessStartInfo(Path.Combine(AppContext.BaseDirectory, "oras.exe")),
+            };
+
+            process.StartInfo.Arguments = command;
+            process.StartInfo.RedirectStandardError = true;
+            process.StartInfo.WorkingDirectory = orasWorkingDirectory;
+            process.EnableRaisingEvents = true;
+
+            // Add event to make it async.
+            process.Exited += new EventHandler((sender, e) => { eventHandled.TrySetResult(true); });
+            try
+            {
+                process.Start();
+            }
+            catch (Exception ex)
+            {
+                throw new OrasException(TemplateManagementErrorCode.OrasProcessFailed, "Oras process failed", ex);
+            }
+
+            StreamReader errStreamReader = process.StandardError;
+            await Task.WhenAny(eventHandled.Task, Task.Delay(30000));
```

Comment:
We can define it (30000) into a variable, so if we want to update it, we can find it easily.